lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Ok. With skipping prototype chain lookup when all prototype objects have no enumerable objects, the perf impact is flat now. But this makes the lookup very easy to be affected by any change in the prototype chain, like a minor change to any prototype object (JS framework is likely to do this?) will fallback almost all prototype to full lookup but this looks more conservative. 
Which option do you prefer, this option or the optimization I suggested on the existing ineffectiveness?